syntax = "proto3";

import "google/protobuf/any.proto";

package heph.ipc.zenoh.proto;

enum ActionServerRequestStatus {
    SUCCESSFUL = 0;
    REJECTED_USER = 1;
    REJECTED_ALREADY_RUNNING = 2;
    INVALID = 3;
    STOPPED = 4;
}

message ActionServerRequestResponse {
    ActionServerRequestStatus status = 1;
};

message ActionServerResponse {
    ActionServerRequestStatus status = 1;
    google.protobuf.Any value = 2;
}
